### DISCUSSION TITLE
Publish confirmation channel is throwing tcp: missing address

### DISCUSSION THREAD

**Discussion by govindkailas** [rabbitmq/rabbitmq-stream-go-client#226] [Q&A] [answered]
I am trying the `getting_started.go` but it seems to be failing at https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/9b8f809347ef825a0ca960c0c601648746ca2a3d/examples/getting_started.go#L77 with `dial tcp: missing address exit status 1`. I can see that the stream is getting created with the UUID.
![image](https://github.com/rabbitmq/rabbitmq-stream-go-client/assets/6698224/30d99a51-bfd6-4544-bce5-687d293fd09f)

I added some extra `Println` to see where it is breaking. 

```
 go run .
Getting started with Streaming client for RabbitMQ
Connecting to RabbitMQ streaming ...
Stream Declared! --> e12772bc-46b2-488d-8b0c-4fe282697ac6
Created a new producer for the Stream --> e12772bc-46b2-488d-8b0c-4fe282697ac6
dial tcp: missing address exit status 1
```
RabbitMQ instance is created using the operator as mentioned [here](https://github.com/govindkailas/rabbitmq-operator-demo/blob/0b01c6ba815262dedfa4fb914f1cf23dc5d7f1a0/rmq.yaml)  and the service is exposed as a Loadbalancer.

Any clues on why it is complaining about the missing address?

**Comment by Gsantomaggio** [comment, marked as answer]
you need to configure the load balancer options, see https://github.com/rabbitmq/rabbitmq-stream-go-client 

Load balancer section 

There is an example here:

https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/main/examples/README.md

**Reply by govindkailas** [reply]
Thanks, used the proxy example and it worked like a charm. It would be nice to have a little more description on where and why it failed instead of just saying `dial tcp: missing address exit status 1`

**Reply by Gsantomaggio** [reply]
You are right.
We are working on a total new version of this client.

See https://github.com/rabbitmq/rabbitmq-stream-go-client/discussions/215 

Where the errors are handled in a better way.

**Comment by marcosbis** [comment]
Hi there, 
Im having the same error when trying to connect a cluster
this is my config:
-------
```go
addresses := []string{
		"rabbitmq-stream+tls://guest:guest@my1.sslcloud.server:5551/dev",
		"rabbitmq-stream+tls://guest:guest@my2.sslcloud.server:5551/dev",
		"rabbitmq-stream+tls://guest:guest@my3.sslcloud.server:5551/dev"}



// 	addressResolver := stream.AddressResolver{
//		Host: "my1.sslcloud.server",
//		Port: 5551,
//	}

	env, err := stream.NewEnvironment(
		stream.NewEnvironmentOptions().
		//	SetAddressResolver(addressResolver).
			SetTLSConfig(&tls.Config{InsecureSkipVerify: true}).
			SetUris(addresses))

	CheckErr(err)
	fmt.Println("Connect")

	streamName := "test-streamer"
	err = env.DeclareStream(streamName,
		stream.NewStreamOptions().
			SetMaxLengthBytes(stream.ByteCapacity{}.GB(2)))

	CheckErr(err)
	fmt.Println("Connect1")

	// Get a new producer for a stream
	producer, err := env.NewProducer(streamName, nil)
	CheckErr(err)
  fmt.Println("Continue to consumer")
-----
this will throw:
dial tcp: missing address exit status 1

if I enable the address resolver I dont get any error on creating the producer but the next lines are never triggered, I shouldn't need to add an address resolver when connected direct to the cluster, right?

**Reply by Gsantomaggio** [reply]
The client needs to lookup the rabbitmq nodes name. 

What it the result of:
```
rabbitmqctl cluster_status
```

you should have something like:
```
rabbitmqctl cluster_status
Cluster status of node rabbit@mynodename ...
Basics

Cluster name: rabbit@xxxxxxx
Total CPU cores available cluster-wide: 8

Disk Nodes

rabbit@mynodename
......

```
